reccmp/project/detect.py

```diff
@@ -0,0 +1,356 @@
+import argparse
+import dataclasses
+import logging
+from pathlib import Path
+import typing
+
+import ruamel.yaml
+
+
+logger = logging.getLogger(__file__)
+
+
+class RecCmpProjectException(Exception):
+    pass
+
+
+class RecCmpProjectNotFoundException(RecCmpProjectException):
+    pass
+
+
+class InvalidRecCmpProjectException(RecCmpProjectException):
+    pass
+
+
+class InvalidRecCmpArgumentException(RecCmpProjectException):
+    pass
+
+
+class UnknownRecCmpTargetException(RecCmpProjectException):
+    pass
+
+
+RECCMP_PROJECT_CONFIG = "reccmp-project.yml"
+RECCMP_USER_CONFIG = "reccmp-user.yml"
+RECCMP_BUILD_CONFIG = "reccmp-build.yml"
+
+
+@dataclasses.dataclass
+class RecCmpTarget:
+    target_id: str
+    filename: str
+    source_root: Path
+
+
+@dataclasses.dataclass
+class RecCmpBuiltTarget(RecCmpTarget):
+    original_path: Path
+    recompiled_path: Path
+    recompiled_pdb: Path
+
+
+def verify_target_names(
+    project_targets: dict[str, typing.Any],
+    user_targets: dict[str, typing.Any],
+    build_targets: dict[str, typing.Any],
+):
+    project_keys = set(project_targets.keys())
+    user_keys = set(user_targets.keys())
+    build_keys = set(build_targets.keys())
+    if project_keys - user_keys:
+        logger.warning("User config %s is missing target ids", RECCMP_USER_CONFIG)
+    if user_keys - project_keys:
+        logger.warning(
+            "User config %s contains too many target ids", RECCMP_USER_CONFIG
+        )
+    if project_keys - build_keys:
+        logger.warning("Build config %s is missing target ids", RECCMP_USER_CONFIG)
+    if build_keys - project_keys:
+        logger.warning(
+            "Build config %s contains too many target ids", RECCMP_USER_CONFIG
```

Comment:
Bug: Output `RECCMP_BUILD_CONFIG` instead of `RECCMP_USER_CONFIG` (two instances)

Reply:
This is fixed.